src/rust/engine/options/src/args.rs

```diff
@@ -3,95 +3,140 @@
 
 use std::env;
 
-use super::id::{NameTransform, OptionId, Scope};
+use super::id::{is_valid_scope_name, NameTransform, OptionId, Scope};
```

Comment:
It's probably best to review this as a completely new file, the diffs are not instructive.